core/arch/arm/kernel/tee_ta_manager.c

```diff
@@ -115,6 +116,55 @@ static void set_tee_rs(struct tee_ta_session *tee_rs)
 	thread_set_tsd(tee_rs);
 }
 
+static bool tee_ta_try_set_busy(struct tee_ta_ctx *ctx,
+			struct tee_ta_session *sess)
+{
+	bool rv = false;
+
+	mutex_lock(&tee_ta_mutex);
+
+	/*
+	 * If we're about to set a context of a single instance TA busy
+	 * make sure that we don't have any other single instance TA in the
+	 * call stack to guarantee that we can't deadlock.
+	 */
```

Comment:
I'm open for suggestions. It has to guarantee that we can't deadlock and it would be an advantage if TAs see a consistent behavior.